android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.react.navsdk;
+
+import android.annotation.SuppressLint;
+import android.app.Presentation;
+import android.graphics.Point;
+import android.hardware.display.DisplayManager;
+import android.hardware.display.VirtualDisplay;
+import androidx.annotation.NonNull;
+import androidx.car.app.AppManager;
+import androidx.car.app.CarContext;
+import androidx.car.app.Screen;
+import androidx.car.app.SurfaceCallback;
+import androidx.car.app.SurfaceContainer;
+import androidx.car.app.model.Action;
+import androidx.car.app.model.ActionStrip;
+import androidx.car.app.model.Template;
+import androidx.car.app.navigation.model.NavigationTemplate;
+import androidx.lifecycle.DefaultLifecycleObserver;
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleObserver;
+import androidx.lifecycle.LifecycleOwner;
+import com.facebook.react.bridge.ReadableMap;
+import com.google.android.gms.maps.CameraUpdate;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.libraries.navigation.NavigationViewForAuto;
+import com.google.android.libraries.navigation.StylingOptions;
+
+public abstract class AndroidAutoBaseScreen extends Screen
+    implements SurfaceCallback, INavigationViewController {
+  private static final String VIRTUAL_DISPLAY_NAME = "SampleAppNavScreen";
+
+  private NavigationViewForAuto mNavigationView;
+  private VirtualDisplay mVirtualDisplay;
+  private Presentation mPresentation;
+  protected GoogleMap mGoogleMap;
+  protected boolean mNavigationInitialized = false;
+  private MapViewController mMapViewController;
+
+  private boolean mAndroidAutoModuleInitialized = false;
+  private boolean mNavModuleInitialized = false;
+  private final AndroidAutoBaseScreen screenInstance = this;
+
+  @Override
+  public void setStylingOptions(StylingOptions stylingOptions) {
+    // TODO(jokerttu): set styling to the navigationView
+  }
+
+  public void onNavigationReady(boolean ready) {
+    mNavigationInitialized = ready;
+    invalidate();
```

Comment:
This is to invalidate layout because of the conditional template rendering. It's actually not supposed to be here. I moved it to the example app with comment.